src/main/java/org/mule/extension/ftp/internal/source/FtpDirectoryListener.java

```diff
@@ -209,7 +212,12 @@ public void poll(PollContext<String, FtpFileAttributes> pollContext) {
           return;
         }
 
-        if (!processFile(file, pollContext)) {
+        InputStream contentStream = new ByteArrayInputStream(file.getOutput().getBytes(StandardCharsets.UTF_8));
```

Comment:
You cannot use this class. It will load to memory (or at least try it) any file, regardless of its size. Instead, use the path of the file `file.getOutput()` to invoke whatever the `read` operation is doing.

Reply:
calling read operation now to get the content as Inputstream.